cartridge/invalid-format.lua

```diff
@@ -0,0 +1,68 @@
+local log = require('log')
+
+local illegal_types = {
+    [''] = true,
+    ['n'] = true,
+    ['nu'] = true,
+    ['s'] = true,
+    ['st'] = true,
+}
+
+rawset(_G, '__cartridge_invalid_format_spaces', {})
+
+-- _space trigger.
+local function before_replace(_, tuple)
+    if tuple == nil then return tuple end
+
+    local name = tuple[3]
+    local format = tuple[7]
+
+    for _, field_def in ipairs(format) do
+        if illegal_types[field_def.type] then
+            _G.__cartridge_invalid_format_spaces[name] = true
```

Comment:
Is it verified with `strict.on()` / on a Debug build? I see `rawset()` above, so existence of the global variable is not tracked and so assigning to it in the script mode will raise an error.

The easier way to solve it is to assign the global directly:

```lua
_G.__cartridge_invalid_format_spaces = {}
```

It is allowed on the module level.

The alternative is to make all accesseses via `rawget()`.

(However, please, don't lean on my words and verify it.)

BTW, why not use a module-local variable?

Reply:
@Totktonada 
To the last question - I used it in tests only. So, I can move it under the test flag, that we've used before in our code. 
And all the tests run with `strict.on()`.

Reply:
Hm. Okay. Maybe I don't know something about the strict mode. If it works, disregard my comment.